src/plugins/intel_gpu/src/plugin/transformations/unsqueeze_broadcast_reshape_sdpa_fusion.cpp

```diff
@@ -42,7 +38,7 @@ UnsqueezeBroadcastReshapeSDPAFusion::UnsqueezeBroadcastReshapeSDPAFusion() {
         return rank_equals(4)(output) && consumers_count(1);
     };
 
-    auto input_a_m = any_input(not_reshape);
+    auto input_a_m = any_input();
```

Comment:
@sshlyapn May I know why "not_reshape" is asked here previously?

Reply:
> Seems that you allowed query input’s reshape, and seems that we need to check whether both sdpa_opt and sdpa_micro supports dynamic padded query input. 
E.g., 
Fused QKV gemm => VariadicSplit (crop + optimized out) => reshape  (optimized out) => sdpa query input 
Not quickly sure which model contains such a pattern.
Maybe you can just create a functional test, which has above pattern, and then check the values are correct. 


@yeonbok This relax is an GQA pattern optimizing by removing broadcast nodes from key and value input paths. The sdpa gpu node was in the exec graph already before this optimizing. So correctness of this special case you mentioned should have been assured already.